projects/core/src/main/kotlin/core/Boat.kt

```diff
@@ -21,10 +23,12 @@ class Boat(private val broadcast: Broadcast, private val props: Pair<Propeller,
 
     fun start(io: Scheduler, clock: Scheduler) {
         val speeds = broadcast.valuesOfType(Motion::class.java)
+            .startWith(Motion(0.0, 0.0))
             .map { speed(it) }
+            .onBackpressureBuffer(128, Action0 {  }, BackpressureOverflow.ON_OVERFLOW_DROP_OLDEST)
```

Comment:
This should probably be [`onBackpressureLatest`](http://reactivex.io/RxJava/javadoc/rx/Observable.html#onBackpressureLatest()):

> `onBackpressureLatest` (new in RxJava 1.1) holds on to the most-recently emitted item from the source Observable and immediately emits that item to its observer upon request. It drops any other items that it observes between requests from its observer.

— [ReactiveX - backpressure operators](http://reactivex.io/documentation/operators/backpressure.html)